tests/integration/models/os.spec.ts

```diff
@@ -450,6 +450,49 @@ describe('OS model', function () {
 				const promise = balena.models.os.download('foo-bar-baz');
 				return expect(promise).to.be.rejectedWith('No such device type');
 			}));
+
+		describe('given well formed and malformed balenaOS versions', () => {
+			it('should reject malformed versions with an error', async () => {
+				const versions = [
+					'2.60.1+foo',
+					'2.60.1-foo',
+					'2.60.1.foo',
+					'2.60.1+rev',
+					'2.60.1+rev1+foo',
+					'2.60.1+rev1-foo',
+					'2.60.1+rev1.foo',
+					'2.60.1+rev1.dev.foo',
+					'2.60.1+rev1.prod.foo',
+				];
+				for (const version of [...versions]) {
+					versions.push(`v${version}`);
+				}
+				for (const version of versions) {
+					await expect(
+						balena.models.os.download('raspberry-pi', version),
+					).to.be.rejectedWith(`Invalid balenaOS version format: ${version}`);
+				}
+			});
+
+			it('should accept well formed versions', async () => {
+				const versions = [
+					'0.0.1',
+					'0.0.1.dev',
+					'0.0.1.prod',
+					'0.0.1+rev1',
+					'0.0.1+rev1.dev',
+					'0.0.1+rev1.prod',
+				];
+				for (const version of [...versions]) {
+					versions.push(`v${version}`);
+				}
+				for (const version of versions) {
+					await expect(
+						balena.models.os.download('raspberry-pi', version),
+					).to.be.rejectedWith(`No such version for the device type`);
```

Comment:
This test works, but performs rather poorly: 5 seconds on my laptop. What I really wanted to test is the `normalizeVersion` function in `lib/os/models/os.ts`, but it is not exported. If I had access to that function, the test would run in milliseconds.
**Any suggestions on how I could test `normalizeVersion` directly?** I thought of using [rewire](https://www.npmjs.com/package/rewire), but I am not sure if it would work because of the peculiar SDK arrangement of defining functions inside functions and returning selected internal functions.

Reply:
@pdcastro Looks like `normalizeVersion` can be safely be moved outside of `getOsModel`

Reply:
> Looks like `normalizeVersion` can be safely be moved to the module scope outside of `getOsModel`

This might work in combination with `rewire` (assuming we don't want to export the function). 👍   
Would however there be any side effects in relation to the "webpacking" and in-browser SDK functionality? Hmm `os.download` is not meant to work in browsers anyway...

Reply:
@pdcastro you can move it to `~/utils/device-os-versions.ts`.